docs/source/design.rst

```diff
@@ -0,0 +1,241 @@
+========
+ Design
+========
+
+When a Matplotlib :obj:`~matplotlib.artist.Artist` object in rendered via the `~matplotlib.artist.Artist.draw` method the following
+steps happen (in spirit but maybe not exactly in code):
+
+1. get the data
+2. convert from unit-full to unit-less data
+3. convert the unit-less data from user-space to rendering-space
+4. call the backend rendering functions
+
+..
+   If we were to call these steps :math:`f_1` through :math:`f_4` this can be expressed as (taking
+   great liberties with the mathematical notation):
+
+   .. math::
+
+      R = f_4(f_3(f_2(f_1())))
+
+   or if you prefer
+
+   .. math::
+
+      R  = (f_4 \circ f_3 \circ f_2 \circ f_1)()
+
+   It is reasonable that if we can do this for one ``Artist``, we can build up
+   more complex visualizations by rendering multiple ``Artist`` to the same
+   target.
+
+However, this clear structure is frequently elided and obscured in the
+Matplotlib code base: Step 3 is only present for *x* and *y* like data (encoded
+in the `~matplotlib.transforms.TransformNode` objects) and color mapped data
+(implemented in the `.matplotlib.colors.ScalarMappable` family of classes); the
+application of Step 2 is inconsistent (both in actual application and when it
+is applied) between artists; each ``Artist`` stores it's data in its own way
+(typically as numpy arrays).
+
+With this view, we can understand the `~matplotlib.artist.Artist.draw` methods
+to be very extensively `curried
+<https://en.wikipedia.org/wiki/Curry_(programming_language)>`__ version of
```

Comment:
```suggestion
<https://en.wikipedia.org/wiki/Currying>`__ version of
```

Pretty sure you mean the concept, not the haskell-based language here

Reply:
Yes, the down side of writing while having insomnia 🤦🏻